### PR TITLE
rawkv: Fix unstable test-region-merge

### LINKED ISSUE #13582: test: cases::test_rawkv::test_region_merge is unstable
### What version of TiKV are you using?

master (1a9446f334c2d29417f656fa8ef4ec6a1dda95f7)

### What operating system and CPU are you using?

CI env.

### Steps to reproduce

Run CI.

### What did you expect?

`cases::test_rawkv::test_region_merge` succeed.

### What did happened?

`cases::test_rawkv::test_region_merge` failed.
```
[2022-10-10T06:25:34.453Z] --- TRY 3 STDERR:                              tests::failpoints cases::test_rawkv::test_region_merge ---
[2022-10-10T06:25:34.453Z] thread 'cases::test_rawkv::test_region_merge' panicked at 'region merge failed', tests/failpoints/cases/test_rawkv.rs:118:17
[2022-10-10T06:25:34.453Z] stack backtrace:
[2022-10-10T06:25:34.453Z]    0: std::panicking::begin_panic
[2022-10-10T06:25:34.453Z]    1: failpoints::cases::test_rawkv::TestSuite::must_merge_region_by_key
[2022-10-10T06:25:34.453Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_rawkv.rs:118:17
[2022-10-10T06:25:34.453Z]    2: failpoints::cases::test_rawkv::test_region_merge
[2022-10-10T06:25:34.453Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_rawkv.rs:246:9
[2022-10-10T06:25:34.453Z]    3: failpoints::cases::test_rawkv::test_region_merge::{{closure}}
[2022-10-10T06:25:34.453Z]              at /home/jenkins/tikv-src/tests/failpoints/cases/test_rawkv.rs:205:1
[2022-10-10T06:25:34.453Z]    4: core::ops::function::FnOnce::call_once
[2022-10-10T06:25:34.453Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/core/src/ops/function.rs:248:5
[2022-10-10T06:25:34.453Z]    5: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}
[2022-10-10T06:25:34.453Z]    6: core::ops::function::FnOnce::call_once{{vtable.shim}}
[2022-10-10T06:25:34.453Z]    7: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
[2022-10-10T06:25:34.453Z]              at /rustc/0f4bcadb46006bc484dad85616b484f93879ca4e/library/alloc/src/boxed.rs:1935:9
[2022-10-10T06:25:34.453Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

Refer to:
https://ci.pingcap.net/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/27741/pipeline/
https://ci.pingcap.net/blue/organizations/jenkins/tikv_ghpr_test/detail/tikv_ghpr_test/27748/pipeline/

### PR DESCRIPTION
**Pull request by pingyu**
Signed-off-by: pingyu <yuping@pingcap.com>

### What is changed and how it works?

Issue Number: Close #13582

What's Changed:

```commit-message
Fix unstable test `test_rawkv::test_region_merge`.
```

The failure of `test_rawkv::test_region_merge` should be caused by the fail point `causal_ts_provider_flush`, which make `handle_update_max_timestamp` in a busy loop and exhaust the pd worker of raftstore.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
No.
- Need to cherry-pick to the release branch
No.

### Check List 

Tests 

- Unit test
- Integration test

Side effects
- No.

### Release note 

```release-note
Fix unstable test `test_rawkv::test_region_merge`.
```

### CODE REVIEW

**Comment by pingyu** [comment]
> /cc @haojinming @zeminzhou

PTAL, thanks~

File: tests/failpoints/cases/test_rawkv.rs [hunk 1]
```diff
@@ -164,8 +164,8 @@ fn test_leader_transfer() {
         assert_eq!(suite.must_raw_get(key1), Some(b"v4".to_vec()));
     }
 
-    // Disable CausalObserver::flush_timestamp to produce causality issue.
-    fail::cfg(FP_CAUSAL_TS_PROVIDER_FLUSH, "return").unwrap();
+    // Make causal_ts_provider.async_flush() & handle_update_max_timestamp fail.
```
Comment (TonsnakeLin) [inline tests/failpoints/cases/test_rawkv.rs:167]: Could you explain why failpoint `FP_GET_TSO` cant make `causal_ts_provider.async_flush() & handle_update_max_timestamp` fail ?

Comment (pingyu) [reply tests/failpoints/cases/test_rawkv.rs:167]: The critical path of `causal_ts_provider.async_flush()` is getting a batch of TSOs from PD and renew the TSO cache. So the error of `get_tso` must make `causal_ts_provider.async_flush()` fail.
Please refer to [`BatchTsoProvider::renew_tso_batch_impl()`](https://github.com/tikv/tikv/blob/1a9446f334c2d29417f656fa8ef4ec6a1dda95f7/components/causal_ts/src/tso.rs#L419).

And `handle_update_max_timestamp()` invokes `causal_ts_provider.async_flus()` [here](https://github.com/tikv/tikv/blob/1a9446f334c2d29417f656fa8ef4ec6a1dda95f7/components/raftstore/src/store/worker/pd.rs#L1622) (when `storage.api-version = 2`). So `handle_update_max_timestamp()` will also fail (and keep retrying until succeed).

**Review by haojinming — APPROVED** [review]
LGTM

**Review by zeminzhou — APPROVED** [review]
LGTM~

**Review by BusyJay — APPROVED** [review]

**Pull request merged**